3_visualize/src/ploting_func.R

```diff
@@ -43,5 +43,6 @@ plot_func <- function (data,out_dir) {
   points(2.2, 1.09, col = '#1b9e77', pch = 21, bg = 'white', lwd = 2.5, cex = 1.5)
   text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
   
+  return(out_file_path)
   dev.off()
```

Comment:
_(this is a suggestion for updating but is not necessary since your pipeline functions)_ The `return()` function should be called as the last step in this function because as soon as you call it, the function will exit meaning the `dev.off()` function is never called right now. So, switching the `return(out_file_path)` to come _after_ the `dev.off()` would resolve this issue.